### PR TITLE
feat: displays added events - admin user

### PR DESCRIPTION
**Pull request by vivianamck**
## Issue ticket number and link
https://trello.com/c/sITZm1Uf

## Checklist before requesting a review
- [X] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

## Pull request type

Please check the type of change your PR introduces:

- [X] :sparkles: Feature
- [ ] :bug: Bugfix
- [ ] :pencil2: Fixing typos
- [ ] :lipstick: :art: Code style update (formatting, renaming)
- [ ] :recycle: Refactoring (no functional changes, no api changes)
- [ ] :zap: Performance
- [ ] :pencil: Documentation content changes
- [ ] :white_check_mark: Unit testing
- [ ] :ok_hand: Updating due to code review suggest
- [ ] :rocket: Release: v
- [ ] :wrench: Add or update configuration files
- [ ] :hammer: Add or update development scripts
- [ ] :bricks: Infrastructure related changes

## What is the current module?

-  Displays events on cards - all users

## What is the new behavior?

- Added page displaying created events
- Added component for table with fields: Name, Date, Number of subscriptions, Status, Actions, filters for the table and button to create event.
- Added component for private route
- Created unit tests for table and component rendering
- Modified auth provider and auth reducer to get value from 'isadmin'

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

image
image

### CODE REVIEW

File: src/components/AppRoutes/ProtectedRoute/ProtectedRoutes.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    user: boolean;
+    component: React.FC;
+}
+
+const ProtectedRoutes: FC<PropType> = ({user, component: Component}) => {
+    if (user) {
+        return <Component />
+    } else {
+        return null
+    }
```
Comment (jlora-mck) [inline src/components/AppRoutes/ProtectedRoute/ProtectedRoutes.tsx:13]: This looks cleaner! 
```suggestion
    if (!user) return null:
    return <Component />
```

File: src/components/EventsAdminView/EventAdminView.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import { render, screen } from '@testing-library/react'
+import EventAdminView, { EventsAdminViewProps } from './EventAdminView'
+
+describe('Description', () => {
+})
```
Comment (jlora-mck) [inline src/components/EventsAdminView/EventAdminView.spec.tsx:5]: Please remove

File: src/components/EventsAdminView/EventAdminView.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+const renderComponent = (props: EventsAdminViewProps) =>
+  render(<EventAdminView  {...props}/>)
```
Comment (jlora-mck) [inline src/components/EventsAdminView/EventAdminView.spec.tsx:8]: if you dont want to use brackets lets go for this notation, the other is a bit misleading
```suggestion
const renderComponent = (props: EventsAdminViewProps) => render(<EventAdminView  {...props}/>)
```

File: src/components/EventsAdminView/EventAdminView.tsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    return dateObject.format('D MMM YYYY')
+  }
+
+  //TODO: Create generic utility(used in components multiple)
+  const handleChangeFilters = (filters: ConferenceFilters) => {
+    if (filters.sortBy) {
+      const sortedAllEvents =
+        filters.sortBy === 'newest'
+          ? JSON.parse(JSON.stringify(events)).sort(sortDescending)
+          : JSON.parse(JSON.stringify(events)).sort(sortAscending)
```
Comment (jlora-mck) [inline src/components/EventsAdminView/EventAdminView.tsx:64]: Why do we need to JSON.parse and JSON.stringify here ? are we not receiving this as an object from the back? Also lets not repeat code

File: src/components/EventsAdminView/EventAdminView.tsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+  loadingHeadquarters: boolean
+  selectedHeadquarter?: string
+}
+
+export default function EventAdminView({
+  events,
+  allHeadquarters,
+  loadingEvents,
+  loadingHeadquarters,
+  selectedHeadquarter = '-1',
```
Comment (jlora-mck) [inline src/components/EventsAdminView/EventAdminView.tsx:48]: why not null ?

File: src/components/EventsAdminView/EventAdminView.tsx [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+  //TODO: Create generic utility(used in components multiple)
+  const handleHeadquarterChanged = (selectedHeadquarter: string) => {
+    let filteredByHeadquarter: Conference[] = JSON.parse(
+      JSON.stringify(events)
+    )
+
+    if (selectedHeadquarter !== '-1') {
+      filteredByHeadquarter = filteredByHeadquarter.filter(
+        (element: Conference) =>
+          element.headquarter && element.headquarter.id === selectedHeadquarter
```
Comment (jlora-mck) [inline src/components/EventsAdminView/EventAdminView.tsx:80]:
```suggestion
        ({ headquarter,  headquarter }: Conference) =>
          headquarter && headquarter.id === selectedHeadquarter
```

File: src/components/EventsAdminView/EventAdminView.tsx [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+                    <TableCell align="center">Status</TableCell>
+                    <TableCell align="center">Actions</TableCell>
+                  </TableRow>
+                </TableHead>
+
+                <TableBody>
+                  {filteredEvents.map((row) => (
+                    <TableRow
+                      key={row.id}
+                      sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
```
Comment (jlora-mck) [inline src/components/EventsAdminView/EventAdminView.tsx:133]: Add this to a style

File: src/contexts/Auth/AuthProvider.tsx [hunk 8]
```diff
@@ -12,10 +12,11 @@ interface VerifyApiResponse {
   userUid: string
   email: string
   username?: string | null
+  isadmin?: boolean | null
 }
 
 export const AuthProvider = ({ children }: Props) => {
-  let initialState = { isAuth: false, userUid: '', email: '', username: '' }
+  let initialState = { isAuth: false, userUid: '', email: '', username: '', isadmin: false }
```
Comment (jlora-mck) [inline src/contexts/Auth/AuthProvider.tsx:19]:
```suggestion
  let initialState = { isAuth: false, userUid: '', email: '', username: '', isAdmin: false }
```

File: src/contexts/Auth/AuthProvider.tsx [hunk 9]
```diff
@@ -56,9 +57,10 @@ export const AuthProvider = ({ children }: Props) => {
     try {
       const userResponse = await usersApi.getUserById(userUid)
```
Comment (jlora-mck) [inline src/contexts/Auth/AuthProvider.tsx:58]:
```suggestion
      const { firstName, isAdmin } = await usersApi.getUserById(userUid)
```

File: src/contexts/Auth/AuthProvider.tsx [hunk 10]
```diff
@@ -56,9 +57,10 @@ export const AuthProvider = ({ children }: Props) => {
...
       dispatch({
-        type: 'UPDATE_USERNAME',
+        type: 'UPDATE_USER',
         payload: {
           username: userResponse.firstName as string,
+          isadmin: userResponse.isAdmin as boolean,
```
Comment (jlora-mck) [inline src/contexts/Auth/AuthProvider.tsx:63]:
```suggestion
          username: firstName as string,
          isAdmin,
```

File: src/pages/EventsAdmin/EventsAdmin.tsx [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+      .then((headquarters) => {
+        setAllHeadquarters(headquarters)
+      })
+      .catch((error) => {
+        console.log('Error retrieving all headquarters')
+        console.error(error)
+      })
+      .finally(() => {
+        setLoadingHeadquarters(false)
+      })
```
Comment (jlora-mck) [inline src/pages/EventsAdmin/EventsAdmin.tsx:30]: Please use async/await like the rest of the app

**Review by jlora-mck — CHANGES_REQUESTED** [review]
Great work! i added some comments... please there files that needs an enter at the end (the ones that have a red circle at the end of the file), Also didnt see eventsAdmin spec, did we forget it ? @vivianamck

File: src/components/AppRoutes/ProtectedRoute/ProtectedRoutes.tsx (hunk 1 above)
Comment (vivianamck) [reply src/components/AppRoutes/ProtectedRoute/ProtectedRoutes.tsx]: Done

File: src/components/EventsAdminView/EventAdminView.tsx (hunk 5 above)
Comment (vivianamck) [reply src/components/EventsAdminView/EventAdminView.tsx]: Done

File: src/contexts/Auth/AuthProvider.tsx (hunk 8 above)
Comment (vivianamck) [reply src/contexts/Auth/AuthProvider.tsx]: Done

File: src/contexts/Auth/AuthProvider.tsx (hunk 10 above)
Comment (vivianamck) [reply src/contexts/Auth/AuthProvider.tsx]: Done

File: src/pages/EventsAdmin/EventsAdmin.tsx
[the remark this answers is not in the document]
Comment (vivianamck) [reply src/pages/EventsAdmin/EventsAdmin.tsx:1]: Done

File: src/components/EventsAdminView/EventAdminView.tsx (hunk 7 above)
Comment (vivianamck) [reply src/components/EventsAdminView/EventAdminView.tsx]: Done

File: src/contexts/Auth/AuthProvider.tsx (hunk 9 above)
Comment (vivianamck) [reply src/contexts/Auth/AuthProvider.tsx]: Done

File: src/pages/EventsAdmin/EventsAdmin.tsx (hunk 11 above)
Comment (vivianamck) [reply src/pages/EventsAdmin/EventsAdmin.tsx]: Done

**Review by jlora-mck — APPROVED** [review]
Nice Work! We can improve the test but we can do it in a separated ticket

**Review by FelipeLosada1996 — APPROVED** [review]

**Review by Sofiag8 — APPROVED** [review]

**Pull request merged**